sqlmesh/core/console.py

```diff
@@ -816,15 +827,29 @@ def start_promotion_progress(
             self.environment_naming_info = environment_naming_info
             self.default_catalog = default_catalog
 
-    def update_promotion_progress(self, snapshot: SnapshotInfoLike, promoted: bool) -> None:
+    def update_promotion_progress(
+        self,
+        snapshot: SnapshotInfoLike,
+        promoted: bool,
+        snapshots_with_virtual_views: t.List[SnapshotId],
+    ) -> None:
         """Update the snapshot promotion progress."""
-        if self.promotion_progress is not None and self.promotion_task is not None:
+        if (
+            self.promotion_progress is not None
+            and self.promotion_task is not None
+            and snapshot.snapshot_id in snapshots_with_virtual_views
```

Comment:
So if the given `snapshot` is not in `snapshots_with_virtual_views` then why call `update_promotion_progress` at all? Shouldn't we do this check upstream?